exec/cnex/cell.c

```diff
@@ -523,16 +523,24 @@ Cell *cell_arrayIndexForWrite(Cell *c, size_t i)
         // Assign the (new) copied array to the cell object.
         c->array = wa;
     }
-    if (i >= c->array->size) {
-        c->array->data = realloc(c->array->data, sizeof(Cell) * (i+1));
+    if (i >= c->array->max) {
+        // ToDo: Fix this!  All array memory management should happen in array.c.
```

Comment:
This needs to be fixed before it can be accepted.

Reply:
Fixed.